src/lib/rules/html-checker/html-checker.ts

```diff
@@ -100,7 +102,15 @@ const rule: IRuleBuilder = {
             const htmlChecker = require('html-validator');
 
             scanOptions.data = response.body.content;
-            htmlCheckerPromise = scanOptions.data ? htmlChecker(scanOptions) : Promise.resolve({messages: []});
+            htmlCheckerPromise = scanOptions.data ? htmlChecker(scanOptions) : Promise.resolve({ messages: [] });
+            htmlCheckerPromise.catch((error) => {
+                failError = error;
```

Comment:
why don't you notify the error here, and then just `return` in the `end` method?